artifactory/commands/generic/download.go

```diff
@@ -295,30 +315,36 @@ func createLegalPath(root, path string) string {
 }
 
 func createSyncDeletesWalkFunction(tempRoot string) gofrog.WalkFunc {
-	return func(path string, info os.FileInfo, err error) error {
+	return func(path string, info os.FileInfo, err error) (currentErr error) {
```

Comment:
The function does not use defers. I think there is no need to use `currentErr` here